server/graphql/v1/TransactionInterface.js

```diff
@@ -297,13 +297,13 @@ export const TransactionExpenseType = new GraphQLObjectType({
             : null;
         },
       },
-      expenseId: {
-        type: GraphQLInt,
+      expense: {
+        type: ExpenseType,
         resolve(transaction, args, req) {
           // If it's a sequelize model transaction, it means it has the method getExpenseFromViewer
           // otherwise we return null
-          return transaction.getExpenseForViewer
-            ? transaction.getExpenseForViewer(req.remoteUser).then(expense => expense && expense.id)
+          return transaction.getExpense
```

Comment:
Let's keep `getExpenseForViewer` here, no?

Reply:
I think `getExpenseForViewer` is better for the next pr that will wrap the other attributes, or we change it for `getExpense` for now?

Reply:
Let's use `getExpenseForViewer`, I think it's what we want now, and for the future.

Reply:
@znarf done.

Reply:
Sorry, I just checked with @Betree and he confirmed he would like to use Dataloader as it was originally written in the spec. Instead of `getExpense` /  `getExpenseForViewer`.

1) introduce an expense dataloader `expense.findById` similar to https://github.com/opencollective/opencollective-api/blob/master/server/graphql/loaders.js#L56
2) use it, ie: `return req.loaders.expense.findById.load(transaction.ExpenseId);`

Reply:
@znarf Done.